tutor/commands/jobs.py

```diff
@@ -0,0 +1,254 @@
+    superuser: str, staff: bool, username: str, email: str, password: str
+) -> str:
+    opts = ""
+    if superuser:
+        opts += " --superuser"
+    if staff:
+        opts += " --staff"
+    return f"""
+./manage.py lms manage_user {opts} {username} {email}
+./manage.py lms shell -c "
+from django.contrib.auth import get_user_model
+u = get_user_model().objects.get(username='{username}')
+u.set_password('{password}')
+u.save()"
+"""
+
+
+@click.command(help="Import the demo course")
+def importdemocourse() -> t.Iterable[t.Tuple[str, str]]:
+    template = """
+# Import demo course
+git clone https://github.com/openedx/edx-demo-course --branch {{ OPENEDX_COMMON_VERSION }} --depth 1 ../edx-demo-course
+python ./manage.py cms import ../data ../edx-demo-course
+
+# Re-index courses
+./manage.py cms reindex_course --all --setup"""
+    yield ("cms", template)
+
+
+@click.command(
+    help="Assign a theme to the LMS and the CMS. To reset to the default theme , use 'default' as the theme name."
+)
+@click.option(
+    "-d",
+    "--domain",
+    "domains",
+    multiple=True,
+    help=(
+        "Limit the theme to these domain names. By default, the theme is "
+        "applied to the LMS and the CMS, both in development and production mode"
+    ),
+)
+@click.argument("theme_name")
+def settheme(domains: t.List[str], theme_name: str) -> t.Iterable[t.Tuple[str, str]]:
+    yield ("lms", set_theme_template(theme_name, domains))
+
+
+def set_theme_template(theme_name: str, domain_names: t.List[str]) -> str:
+    """
+    For each domain, get or create a Site object and assign the selected theme.
+    """
+    # Note that there are no double quotes " in this piece of code
+    python_command = """
+import sys
+from django.contrib.sites.models import Site
+def assign_theme(name, domain):
+    print('Assigning theme', name, 'to', domain)
+    if len(domain) > 50:
+            sys.stderr.write(
+                'Assigning a theme to a site with a long (> 50 characters) domain name.'
+                ' The displayed site name will be truncated to 50 characters.\\n'
+            )
+    site, _ = Site.objects.get_or_create(domain=domain)
+    if not site.name:
+        name_max_length = Site._meta.get_field('name').max_length
+        site.name = domain[:name_max_length]
+        site.save()
+    site.themes.all().delete()
+    site.themes.create(theme_dir_name=name)
+"""
+    domain_names = domain_names or [
+        "{{ LMS_HOST }}",
+        "{{ LMS_HOST }}:8000",
+        "{{ CMS_HOST }}",
+        "{{ CMS_HOST }}:8001",
+        "{{ PREVIEW_LMS_HOST }}",
+        "{{ PREVIEW_LMS_HOST }}:8000",
+    ]
+    for domain_name in domain_names:
+        python_command += f"assign_theme('{theme_name}', '{domain_name}')\n"
+    return f'./manage.py lms shell -c "{python_command}"'
+
+
+hooks.Filters.CLI_DO_COMMANDS.add_items(
+    [
+        createuser,
+        importdemocourse,
+        initialise,
+        settheme,
+    ]
+)
```

Comment:
Ugh, I just banged my head against my keyboard for half an hour before realizing why this wasn't working:
```python
def myjob():
    return [("lms", "echo hello world")]
hooks.Filters.CLI_DO_COMMANDS.add(myjob)
```
(can you see it?)
Other than the strongly typed hooks PR you have open, I'm not sure what else we would do to save a plugin developer from this kind of mistake.

Reply:
`add_item`? To your credit, yes I must admit it's a little confusing. We'll see whether typed hooks help resolve this situation.

Reply:
I think it will.